control/eiger/eiger_detector.py

```diff
@@ -538,7 +540,13 @@ def write_detector_config(self, item, value):
     def read_detector_status(self, item):
         if item == 'stale_parameters':
             # Read stale parameters flag
-            response = {u'value': len(self._stale_parameters) != 0, u'access_mode': u'r', u'value_type': u'bool'}
+            with self._lock:
+                response = {
+                    u'value': len(self._stale_parameters) != 0,
+                    u'access_mode': u'r',
+                    u'value_type':
+                    u'bool'
+                }
```

Comment:
I don't think we need this bit when we have the the set at line 717 as well. Am I missing something @ajgdls ?

Reply:
I added another comment near the line. It is 699 pre diff and 717 including the diff